amr-wind/physics/TerrainDrag.cpp

```diff
@@ -0,0 +1,110 @@
+#include "amr-wind/physics/TerrainDrag.H"
+#include "amr-wind/CFDSim.H"
+#include "AMReX_iMultiFab.H"
+#include "AMReX_MultiFabUtil.H"
+#include "AMReX_ParmParse.H"
+#include "AMReX_ParReduce.H"
+#include "amr-wind/utilities/trig_ops.H"
+#include "amr-wind/utilities/IOManager.H"
+
+namespace amr_wind::terraindrag {
+
+namespace {} // namespace
+
+TerrainDrag::TerrainDrag(CFDSim& sim)
+    : m_sim(sim)
+    , m_repo(sim.repo())
+    , m_mesh(sim.mesh())
+    , m_velocity(sim.repo().get_field("velocity"))
+    , m_terrainBlank(sim.repo().declare_field("terrainBlank", 1, 1, 1))
+    , m_terrainDrag(sim.repo().declare_field("terrainDrag", 1, 1, 1))
+{
+    std::ifstream file("terrain.amrwind");
+    amrex::Real value1, value2, value3;
+    while (file >> value1 >> value2 >> value3) {
```

Comment:
do you need a bunch of sanity checks to make sure the file is formatted as expected?